src/client/caching/stores/BanStore.ts

```diff
@@ -27,19 +27,27 @@ export class BanStore extends DataStore<Ban> {
 	}
 
 	/**
-	 * Adds a new structure to this DataStore
-	 * @param data The data packet to add
+	 * Bans a user, and optionally delete previous messages sent by the banned user.
+	 * @since 0.0.1
+	 * @param userID The {@link User user} ID to ban from the server.
+	 * @see https://discord.com/developers/docs/resources/guild#create-guild-ban
 	 */
-	// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
-	// @ts-ignore
-	protected _add(data: GuildBanAddDispatch['d']): Ban {
-		const existing = this.get(data.user.id);
-		// eslint-disable-next-line dot-notation
-		if (existing) return existing['_patch']();
+	public async add(userID: string, options: BanAddOptions = {}): Promise<this> {
+		const endpoint = Routes.guildBan(this.guild.id, userID);
+		await this.client.api.put(endpoint, { query: { 'delete-message-days': options.deleteMessageDays, reason: options.reason } });
```

Comment:
You sure you're supposed to do reason in query and not header?

Reply:
Ban reason !== Audit log reason.